src/main.c

```diff
@@ -1393,6 +1393,27 @@ static yoml_t *resolve_file_tag(yoml_t *node, resolve_tag_arg_t *arg)
     return loaded;
 }
 
+static yoml_t *resolve_env_tag(yoml_t *node, resolve_tag_arg_t *arg)
+{
+
+    const char * env_var = node->data.scalar;
+    if (node->type != YOML_TYPE_SCALAR) {
+        fprintf(stderr, "value of !env must be a scalar");
+        return NULL;
+    }
+
+    const char *env_var_content;
+    if ((env_var_content = getenv(env_var)) == NULL)
+        env_var_content = "";
+
+    free(node->data.scalar);
+    node->data.scalar = h2o_strdup(NULL, env_var_content, SIZE_MAX).base;
```

Comment:
what is the reason the pointer from `getenv` cant be assigned directly to `node->data.scalar` ?

Reply:
Because `node->data.scalar` eventually gets `free`d in `yoml_free`.